### PR TITLE
ADR 36: NextJS App base paths

### PR DESCRIPTION
**Pull request by eatyourgreens**
Adding base paths to the NextJS apps, so that we can host them both on the same domain.

Package:
app-content-pages
app-project

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: docs/arch/adr-36.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+## Context
+
+When we launched the NextJS apps, a single domain could only host one NextJS app. We gave each app its own subdomain: `fe-content-pages.zooniverse.org` and `fe-projects.zooniverse.org` then proxied URLs from `www.zooniverse.org` to those domains. Next data requests, on the `/_next` URL, are proxied from `www.zooniverse.org/_next` to `fe-project.zooniverse.org/_next`. We deliberately broke the content pages app, in favour of supporting projects.
+
+[Next 9.5](https://nextjs.org/blog/next-9-5), in July 2020, added support for multiple apps running on the same domain via the [`basePath`](https://nextjs.org/blog/next-9-5#customizable-base-path) config setting.
+
+## Decision
+
+Set base paths of `/about` and `/projects` to the content pages app and project app respectively.
```
Comment (eatyourgreens) [inline docs/arch/adr-36.md:13]:
```suggestion
Set base paths of `/about` and `/projects` for the content pages app and project app respectively.
```

File: docs/arch/adr-36.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+[Next 9.5](https://nextjs.org/blog/next-9-5), in July 2020, added support for multiple apps running on the same domain via the [`basePath`](https://nextjs.org/blog/next-9-5#customizable-base-path) config setting.
+
+## Decision
+
+Set base paths of `/about` and `/projects` for the content pages app and project app respectively.
+https://github.com/zooniverse/front-end-monorepo/pull/2519
+
+## Consequences
+
```
Comment (eatyourgreens) [inline docs/arch/adr-36.md:17]: A big consequence is that data requests to `/_next` can be supported for both apps, where they’re not supported for content pages at the moment.

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Documentation looks good, and basic concepts (as a result of #2519 ) seems understandable . LGTM 👍

File: docs/arch/adr-36.md (hunk 2 above)
Comment (eatyourgreens) [reply docs/arch/adr-36.md:17]:
```suggestion
 - data requests to `/_next` can now be supported for both apps, at `/about/_next` and `/projects/_next`. 
```

File: docs/arch/adr-36.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- Static assets and compiled JS can be served via `www.zooniverse.org/about/_next` and `www.zooniverse.org/projects/_next`, taking advantage of our CDN.
+- `/about` and `/projects` are removed from the page routes in the NextJS apps, so internal links and redirects have had to be updated throughout the code.
+- The root URLs of the apps, for status checks and so on, change to https://fe-project.zooniverse.org/projects/ and https://fe-content-pages.zooniverse.org/about/. Requests to /, on either domain, will 404.
+- Our nginx routing has to be configured to pass the `/_next` URLs through to the individual apps, running on kubernetes.
+
+
+## Status
+
+In progress
```
Comment (eatyourgreens) [inline docs/arch/adr-36.md:26]:
```suggestion
Accepted
```

**Pull request merged**